### PR TITLE
Add `validate_df` task to `OutlookToADLS` flow

### PR DESCRIPTION
**Pull request by jkrobicki**
## Summary
- Added `validate_df_dict` parameter to `OutlookToADLS` flow
- Added tests for the flow run and new parameter in the `OutlookToADLS` flow

## Importance
New functionality to check the data

## Checklist

This PR:
- [x] follows the guidelines laid out in `CONTRIBUTING.md`
- [ ] links relevant issue(s)
- [x] adds/updates tests (if appropriate)
- [x] adds/updates docstrings (if appropriate)
- [ ] adds an entry in `CHANGELOG.md`

### CODE REVIEW

File: viadot/flows/outlook_to_adls.py [hunk 1]
```diff
@@ -29,6 +30,7 @@ def __init__(
         limit: int = 10000,
         timeout: int = 3600,
         if_exists: Literal["append", "replace", "skip"] = "append",
+        validation_df_dict: dict = None,
```
Comment (gwieloch) [inline viadot/flows/outlook_to_adls.py:33]: `validate_df_dict: dict = None,`

File: viadot/flows/outlook_to_adls.py [hunk 2]
```diff
@@ -54,6 +56,8 @@ def __init__(
             timeout(int, optional): The amount of time (in seconds) to wait while running this task before
                 a timeout occurs. Defaults to 3600.
             if_exists (Literal['append', 'replace', 'skip'], optional): What to do if the local file already exists. Defaults to "append".
+            validation_df_dict (dict, optional): An optional dictionary to verify the received dataframe.
```
Comment (gwieloch) [inline viadot/flows/outlook_to_adls.py:59]: validate_df_dict

File: viadot/flows/outlook_to_adls.py [hunk 3]
```diff
@@ -65,6 +69,9 @@ def __init__(
         self.local_file_path = local_file_path
         self.if_exsists = if_exists
 
+        # Validate DataFrame
+        self.validation_df_dict = validation_df_dict
```
Comment (gwieloch) [inline viadot/flows/outlook_to_adls.py:73]: up

File: viadot/flows/outlook_to_adls.py [hunk 4]
```diff
@@ -98,6 +105,11 @@ def gen_flow(self) -> Flow:
         dfs = apply_map(self.gen_outlook_df, self.mailbox_list, flow=self)
 
         df = union_dfs_task.bind(dfs, flow=self)
+
+        if self.validation_df_dict:
+            validation = validate_df(df=df, tests=self.validation_df_dict, flow=self)
```
Comment (gwieloch) [inline viadot/flows/outlook_to_adls.py:110]: up

**Review by gwieloch — CHANGES_REQUESTED** [review]
pls change new parameter to `validate_df_dict`

File: viadot/flows/outlook_to_adls.py (hunk 1 above)
Comment (jkrobicki) [reply viadot/flows/outlook_to_adls.py]: fixed

File: viadot/flows/outlook_to_adls.py (hunk 2 above)
Comment (jkrobicki) [reply viadot/flows/outlook_to_adls.py]: fixed

File: viadot/flows/outlook_to_adls.py (hunk 3 above)
Comment (jkrobicki) [reply viadot/flows/outlook_to_adls.py]: fixed

File: viadot/flows/outlook_to_adls.py (hunk 4 above)
Comment (jkrobicki) [reply viadot/flows/outlook_to_adls.py]: fixed

**Review by gwieloch — APPROVED** [review]

**Review by m-paz — APPROVED** [review]

**Pull request merged**